src/main/io/gps.c

```diff
@@ -280,6 +278,13 @@ uint16_t gpsConstrainHDOP(uint32_t hdop)
 
 void gpsThread(void)
 {
+    /* Extra delay for at least 2 seconds after booting to give GPS time to initialise */
+    if (isMPUSoftReset() && (millis() < GPS_BOOT_DELAY)) {
```

Comment:
The logic is probably inverted - you want to wait on Hard Reset, not soft one.

This will also cause two second GPS glitch after 50 days uptime - that could be probably ignored, but who knows ...